### PR TITLE
Add version to footer

### PR DESCRIPTION
**Pull request by little9**
Closes #154

### CODE REVIEW

File: config/initializers/git_sha.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+GIT_SHA =
+  if Rails.env.production? && File.exist?('/opt/laevigata/revisions.log')
+    `tail -1 /opt/laevigata/revisions.log`.chomp.split(" ")[3].gsub(/\)$/, '')
```
Comment (bess) [inline config/initializers/git_sha.rb]: I think you mean `/opt/mahonia` here

File: config/initializers/git_sha.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  elsif Rails.env.development? || Rails.env.test?
+    `git rev-parse HEAD`.chomp
+  else
+    "Unknown SHA"
+  end
+
+BRANCH =
+  if Rails.env.production? && File.exist?('/opt/laevigata/revisions.log')
+    `tail -1 /opt/laevigata/revisions.log`.chomp.split(" ")[1]
```
Comment (bess) [inline config/initializers/git_sha.rb]: and here

File: config/initializers/git_sha.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  elsif Rails.env.development? || Rails.env.test?
+    `git rev-parse --abbrev-ref HEAD`.chomp
+  else
+    "Unknown branch"
+  end
+
+LAST_DEPLOYED =
+  if Rails.env.production? && File.exist?('/opt/laevigata/revisions.log')
+    deployed = `tail -1 /opt/laevigata/revisions.log`.chomp.split(" ")[7]
```
Comment (bess) [inline config/initializers/git_sha.rb]: and here :)

**Review by bess — CHANGES_REQUESTED** [review]
Hey, @little9 
Don't forget to change the name of the project. This code depends on reading the deploy data from where capistrano will deploy it. In this case, that will be `/opt/mahonia`

File: config/initializers/git_sha.rb (hunk 1 above)
Comment (no-reply) [reply config/initializers/git_sha.rb]: Should all of these be an ENV variable, perhaps?

File: config/initializers/git_sha.rb (hunk 3 above)
Comment (little9) [reply config/initializers/git_sha.rb]: Sorry! 😳

**Review by bess — CHANGES_REQUESTED** [review]
@little9 Sorry, looks like I didn't flag every instance of `laevigata` that needed to be replaced. @no-reply is right, we should totally replace this with an environment variable.

**Comment by little9** [comment]
@bess @no-reply  this was fun 👹

File: .env [hunk 4]
```diff
@@ -2,4 +2,11 @@ DATACITE_DOMAINS=*
 DATACITE_LOGIN=DATACITE.DCE
 DATACITE_PASSWORD=YOUR_PASSWORD_HERE
 DATACITE_PREFIX=10.24354
-SIDEKIQ_CONCURRENCY=5
\ No newline at end of file
+SIDEKIQ_CONCURRENCY=5
+# Version Information for footer
+GIT_SHA_DEV=$(git rev-parse HEAD 2> /dev/null)
```
Comment (no-reply) [inline .env]: @little9, @bess It feels to me like easy variability across environments should be an advantage of using `ENV` here. Could we just do `GIT_SHA`, `GIT_BRANCH`, and `LAST_DEPLOYED`, then set the variables in deployment?

Sorry to be a bit of a pain about this one. :heart:

Comment (little9) [reply .env]: @bess @no-reply 

I changed this so it's just referencing the variables. I'm not sure about the best way to set them in production. These return the correct values on the server:

```bash
GIT_SHA=$(tail -1 /opt/mahonia/revisions.log | awk '{ print substr($4, 1, length($4)-1) }' 2> /dev/null)
GIT_BRANCH=$(tail -1 /opt/mahonia/revisions.log | awk '{ print $2 }' 2> /dev/null)
LAST_DEPLOYED=$(tail -1 /opt/mahonia/revisions.log | awk '{ print $7 }' | date +%Y-%m-%d 2> /dev/null)
```

**Comment by little9** [comment]
@bess @no-reply 

I changed this so it's just referencing the variables. I'm not sure about the best way to set them in production. These return the correct values on the server:

```bash
GIT_SHA=$(tail -1 /opt/mahonia/revisions.log | awk '{ print substr($4, 1, length($4)-1) }' 2> /dev/null)
GIT_BRANCH=$(tail -1 /opt/mahonia/revisions.log | awk '{ print $2 }' 2> /dev/null)
LAST_DEPLOYED=$(tail -1 /opt/mahonia/revisions.log | awk '{ print $7 }' | date +%Y-%m-%d 2> /dev/null)
```

this is working in my dev environment (except for my brach names which normally have forward slashes): 

```bash
GIT_SHA=$(git rev-parse HEAD 2> /dev/null)
GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
LAST_DEPLOYED_DEV="Not in deployed environment" 
```

**Comment by little9** [comment]
We could put one set of vars in `.env.production` and the other in `.env.development`

**Comment by no-reply** [comment]
I think we should re-add the default settings to the `.env`. So:

```sh
GIT_SHA=$(git rev-parse HEAD 2> /dev/null)
GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
LAST_DEPLOYED_DEV="Not in deployed environment" 
```

It would also be good to get a commit message in before merging this.

**Comment by bess** [comment]
Closing in favor of PR 158

**Pull request closed without merging**